admin-dev/themes/new-theme/js/pages/order/create/order-product-component.js

```diff
@@ -0,0 +1,313 @@
+        this._updateStock(value.stock);
+      }
+
+      $(createOrderPageMap.productSelect).append(
+        $('<option></option>').attr('value', value.product_id).text(name).attr('data-index', index)
+      );
+    }
+
+    const $noRecordsRow = $(createOrderPageMap.noRecordsFound).find(
+      createOrderPageMap.noRecordsFoundRow
+    );
+
+    if ($noRecordsRow.length !== 0) {
+      $noRecordsRow.remove();
+    }
+
+    this._showResultBlock();
+  }
+
+  /**
+   * Updates stock text helper value
+   *
+   * @param stock
+   * @private
+   */
+  _updateStock(stock) {
+    $(createOrderPageMap.inStockCounter).text(stock);
+    $(createOrderPageMap.quantityInput).attr('max', stock);
+  }
+
+  /**
+   * Adds available fields related to selected product
+   *
+   * @param product
+   * @private
+   */
+  _fillRelatedProductFields(product) {
+    this._fillCombinations(product.combinations);
+    this._resolveCustomizationFields(product.customization_fields);
+  }
+
+  /**
+   * Handles product select change
+   *
+   * @param event
+   * @private
+   */
+  _handleProductChange(event) {
+    const index = $(event.target).find(':selected').data('index');
+    const product = this.products[index];
+
+    this._fillRelatedProductFields(product);
+
+    if (product.combinations === null) {
+      this.combinations = [];
+      this._updateStock(product.stock);
+    }
+
+    this._fillCombinations(product.combinations);
+  }
+
+  /**
+   * Handles combination select change
+   *
+   * @param event
+   * @private
+   */
+  _handleCombinationChange(event) {
+    const index = $(event.target).find(':selected').data('index');
+    const combination = this.combinations[index];
+
+    this._updateStock(combination.stock);
+  }
+
+  /**
+   * Fills combination select with options
+   *
+   * @param combinations
+   * @private
+   */
+  _fillCombinations(combinations) {
+    if (combinations === null) {
+      this._removeCombinationSelect();
+
+      return;
+    }
+
+    this.combinations = combinations.combinations;
+
+    if ($(createOrderPageMap.combinationsRow).length === 0) {
+      const $combinationsTemplate = $($(createOrderPageMap.combinationsTemplate).html());
+      $(createOrderPageMap.productSelectRow).after($combinationsTemplate);
+    }
+
+    $(createOrderPageMap.combinationsSelect).empty();
+
+    const entries = Object.entries(combinations.combinations);
+
+    let i = 0; // This is needed because index in this case is attribute combination id
+
+    for (const [id, combination] of entries) {
+      if (i === 0) {
+        this._updateStock(combination.stock);
+      }
+
+      const name = combination.attribute + ' - ' + combination.formatted_price;
+      $(createOrderPageMap.combinationsSelect).append($('<option></option>')
+        .attr('value', id).text(name));
+
+      i += 1;
+    }
+
+    $(createOrderPageMap.combinationsSelect)
+      .on('change', (event) => this._handleCombinationChange(event));
```

Comment:
It is because combination select is not ind DOM if there is no combinations so after rendering it i have to set listener every time. Is there maybe a better way?

Reply:
I think you need to delegate event using `$parent.on('change', '#selector', callback)`